src/main/java/Duke.java

```diff
@@ -1,10 +1,321 @@
+            assert line != null;
+            String[] words = line.split(" ", 2);
+            try{
+                checkValidCommand(words);
+            } catch (InvalidCommandError e) {
+                printDivider();
+                System.out.println("Invalid command. Try again!");
+                printDivider();
+            }
+
+        }
+        return;
+    }
+
+    private static void addCommands() {
+        commands.add("done");
+        commands.add("todo");
+        commands.add("event");
+        commands.add("deadline");
+        commands.add("delete");
+        commands.add("list");
+        commands.add("save");
+        commands.add("bye");
+    }
+    private static void readFile() {
+        try {
+            readFileContents(filePath);
+            System.out.println("Saved tasks successfully loaded!");
+            listAllTask();
+        } catch (FileNotFoundException e) {
+            System.out.println("There is no preloaded file found! Please input your own tasks!");
+            printDivider();
+        }
+    }
+
+    private static void readFileContents (String filePath) throws FileNotFoundException{
+        File f = new File(filePath);
+        Scanner s = new Scanner(f);
+        while (s.hasNext()) {
+            String[] splitInput = s.nextLine().split("\\|");
+            checkFileInputs(splitInput);
+        }
+    }
+
+    public static void saveTasks(String filePath) throws IOException {
+        FileWriter fw = new FileWriter(filePath);
+        String stringToAdd = "";
+        String taskDescription, type, date;
+        Boolean isDone;
+        for(int i = 0; i < tasks.size(); i++) {
+            Task currentTask = tasks.get(i);
+            taskDescription = extractTask(currentTask);
+            type = currentTask.getType();
+            isDone = currentTask.getStatus();
+            stringToAdd +=  type+ "|" + isDone + "|" + taskDescription;
+            if(type == "e" || type == "d") {
+                String[] splitString = currentTask.toString().split(":", 2);
+                String removedSpace = splitString[1].trim();
+                date = removedSpace.replace(")", "");
+                stringToAdd += "|" + date;
+            }
+            stringToAdd += System.lineSeparator();
+        }
+        fw.write(stringToAdd);
+        fw.close();
+        printDivider();
+        System.out.println("Tasks saved successfully");
+        printDivider();
+    }
+
+    private static String extractTask(Task currentTask) {
+        String taskWithDate;
+        String[] taskDescription;
+        taskWithDate = currentTask.toString().split(" ", 2)[1];
+        taskDescription = taskWithDate.split("\\(", 2);
+        return taskDescription[0].trim();
+    }
+
+    private static void checkFileInputs(String[] splitInput) {
+        String type = splitInput[0];
+        Boolean status = Boolean.parseBoolean(splitInput[1]);
+        String taskDescription = splitInput[2];
```

Comment:
nice that you have renamed splitInput[0],[1],[2] in to order to improve readability...